src/NATS.Client/Msg.cs

```diff
@@ -161,11 +180,37 @@ internal MsgHeader(byte[] bytes, int byteCount) : base()
                                 end < byteCount - 1 && bytes[end] != '\r' && bytes[end + 1] != '\n';
                                     end++);
 
-                            Add(Status, Encoding.UTF8.GetString(bytes, start, end - start));
+                            string status = Encoding.UTF8.GetString(bytes, start, end - start);
 
-                            // start other kv pairs at end of the status + \r\n.
+                            // set the start for kv pairs at end of the status + \r\n.
                             kvStart = end + 2;
 
+                            // Fastpath - we only have a code...
+                            if (status.Length == 3)
+                            {
+                                Add(Status, status);
+                                break;
+                            }
+
+                            // We can have code and description.
+
+                            // trim any leading whitespace.
+                            status = status.TrimStart();
+
+                            // check for description.
+                            int spaceIdx = status.IndexOf(' ');
+                            if (spaceIdx <= 0)
```

Comment:
This won't be 0 due to the TrimStart.
```suggestion
                            if (spaceIdx < 0)
```

Reply:
Thanks, sounds good.